.github/tests/pre-install.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -xe
+
+SCRIPT="$(readlink -f "$0")"
+SCRIPTPATH="$(dirname "${SCRIPT}")"
+DEPS="${SCRIPTPATH}/dependencies"
+
+# Set repo and version env variables
+REPOS=$(jq -r '.[] | "export " + ("HELM_REPO_" + .name | ascii_upcase | gsub("-";"_")) + "=" + .repo' "${SCRIPTPATH}/charts.json")
+VERSIONS=$(jq -r '.[] | "export " + ("VERSION_" + .name | ascii_upcase | gsub("-";"_")) + "=" + .version' "${SCRIPTPATH}/charts.json")
+eval "$REPOS"
+eval "$VERSIONS"
```

Comment:
wanted to make things more transparant for the end users. There is so much abstraction and plumbing going on that things aren't as straight forward as I want them to be anymore.

Maybe some follow-up refactors to use a Make native approach I can simplify things further

Reply:
I don't see how it is difficult on users for them to see it source the other file. A comment on what it does when sourcing is probably sufficient? (ex: #include common environment variables for chart repos and versions) It may even make it more readable that way. It also makes it harder on us if we want to update the logic later as we cant change it in one place anymore, and when contributors want to make new ones, they have to copy it yet again.

Reply:
Moved back to central script for now. Till we further refactor and simplify in follow-up PRs.